File: lib/models/barcode.dart
import 'package:mlkit_scanner/mlkit_scanner.dart';

/// Represents a single recognized barcode and its value.
class Barcode {
  /// Barcode value as it was encoded in the barcode.
  final String rawValue;

  /// Barcode value in a user-friendly format.
  final String? displayValue;

  /// Barcode format.
  final BarcodeFormat format;

  /// Format type of the barcode value.
  final BarcodeValueType valueType;

  /// Creates a decoded barcode returned by the native recognizer.
  const Barcode({
    required this.rawValue,
    required this.valueType,
    required this.format,
    this.displayValue,
  });

  /// Creates a barcode from its platform-channel representation.
  factory Barcode.fromJson(Map<String, dynamic> json) {
    return Barcode(
      rawValue: json['raw_value'],
      displayValue: json['display_value'],
      valueType: BarcodeValueTypeCode.fromCode(json['value_type']),
      format: BarcodeFormatCode.fromCode(json['format']),
    );
  }
}

File: android/src/main/kotlin/com/dns_technologies/mlkit_scanner/scanner/models/Barcode.kt
package com.dns_technologies.mlkit_scanner.scanner.models

/**
 * Describes a recognized barcode sent to Dart.
 *
 * @property rawValue Value as it was encoded in the barcode.
 * @property displayValue User-friendly representation of the barcode value, when available.
 * @property format Barcode format code understood by the Dart model.
 * @property valueType Barcode content type code understood by the Dart model.
 */
class Barcode(
    val rawValue: String,
    val displayValue: String?,
    val format: Int,
    val valueType: Int,
) {
    /** Converts this barcode into the Dart platform-channel payload. */
    fun toMap(): Map<String, Any?> = mapOf(
        "raw_value" to rawValue,
        "display_value" to displayValue,
        "format" to format,
        "value_type" to valueType,
    )
}
